src/libs/auth.js

```diff
@@ -8,47 +8,86 @@ import { clearErrorCode, reportError } from 'src/libs/error'
 import * as Utils from 'src/libs/utils'
 
 
-export const getAuthInstance = () => {
+const getAuthInstance = () => {
   return window.gapi.auth2.getAuthInstance()
 }
 
 export const getUser = () => {
-  return getAuthInstance().currentUser.get()
-}
-
-export const getBasicProfile = () => {
-  return getUser().getBasicProfile()
-}
-
-export const getAuthToken = () => {
-  return getUser().getAuthResponse(true).access_token
+  return authStore.get().user
 }
 
 export const signOut = () => {
   sessionStorage.clear()
   getAuthInstance().signOut()
 }
 
+export const reloadAuthToken = () => {
+  return getAuthInstance().currentUser.get().reloadAuthResponse().then(() => true, () => false)
+}
+
 export const authStore = Utils.atom({
   isSignedIn: undefined,
-  registrationStatus: undefined
+  registrationStatus: undefined,
+  user: {}
 })
 
-export const initializeAuth = _.memoize(async () => {
+const initializeAuth = _.memoize(async () => {
   await new Promise(resolve => window.gapi.load('auth2', resolve))
   await window.gapi.auth2.init({ clientId: await Config.getGoogleClientId() })
-  authStore.update(state => ({ ...state, isSignedIn: getAuthInstance().isSignedIn.get() }))
-  getAuthInstance().isSignedIn.listen(isSignedIn => {
+  const processUser = user => {
     return authStore.update(state => {
-      return { ...state, isSignedIn, registrationStatus: isSignedIn ? state.registrationStatus : undefined }
+      const authResponse = user.getAuthResponse(true)
+      const profile = user.getBasicProfile()
+      const isSignedIn = user.isSignedIn()
+      return {
+        ...state,
+        isSignedIn,
+        registrationStatus: isSignedIn ? state.registrationStatus : undefined,
+        user: {
+          token: authResponse && authResponse.access_token,
+          id: user.getId(),
+          email: profile && profile.getEmail(),
+          name: profile && profile.getName(),
+          givenName: profile && profile.getGivenName(),
+          familyName: profile && profile.getFamilyName(),
+          imageUrl: profile && profile.getImageUrl()
+        }
+      }
     })
-  })
+  }
+  processUser(getAuthInstance().currentUser.get())
+  getAuthInstance().currentUser.listen(processUser)
 })
 
+// This is intended for tests to short circuit the login flow
+window.forceSignIn = async token => {
+  const res = await fetch(
```

Comment:
For the await, it's a bit arbitrary. The line was already long, I suppose. 

For the data, I usually lean this way when there's a large number of fields being extracted in bulk, such that the destructuring could get large. Especially since we're renaming several of them anyway, this felt a bit clearer.

Reply:
Ah. I find the renaming destructure to be easier, but this really isn't a big deal.